tools/serverpod_cli/lib/src/generator/dart/library_generators/library_generator.dart

```diff
@@ -2174,3 +2204,12 @@ extension on Iterable<SerializableModelFieldDefinition> {
     }
   }
 }
+
+extension on SerializableModelDefinition {
+  /// Prevent generated future call models from being exported
+  /// from generated protocol code. This ensures that only
+  /// user defined models are exported.
+  bool get shouldExport {
+    return !RegExp(r'^future_calls_models\/.*').hasMatch(fileName);
```

Comment:
Suggestion: To reduce the change of collision with user-defined folders, it would be good to add a `generated` word in this path:

```suggestion
    return !RegExp(r'^future_calls_generated_models\/.*').hasMatch(fileName);
```

And we should not forget to mention this protected path when documenting the new interface after this PR is merged.